### PR TITLE
add Rustlang drag-race version

### PR DESCRIPTION
**Pull request by joshallen64**
I saw there was 3 or so Rust implementations, but I thought I'd throw mine in here to see how it compares to the others.

### CODE REVIEW

**Comment by rbergen** [comment]
@joshallen64 We're still happy to consider this solution, but some work will be required to make it possible to add it.

If you still want this solution to be included in the drag race then please:
- Read the [contributing guidelines](https://github.com/davepl/Primes/blob/drag-race/CONTRIBUTING.md) and structure your solution accordingly.
- Consider that there is an open PR that targets the PrimeRust/solution_3 directory. To prevent a merge conflict later, please use PrimeRust/solution_4, for yours.

Due to the large amount of changes to the `drag-race` branch since you opened this PR, it is probably best to base your next commit on the current version of this branch.

**Comment by joshallen64** [comment]
@rbergen Thanks, it looks like a lot has changed!  I'll update it to the new contribution guidelines and retarget for solution_4.

**Comment by joshallen64** [comment]
@rbergen  I believe I have updated everything to conform with the contribution guidelines.  Let me know if there are any issues.

File: PrimeRust/solution_4/PrimeRust.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+                break
+            }
+        }
+
+        if show_results {
+            println!("");
+        }
+        println!("joshallen64;{};{:?};1",
+                passes,
+                duration
```
Comment (rbergen) [inline PrimeRust/solution_4/PrimeRust.rs]: This prints the duration including an "s" at the end, My guess is that you'll have to explicitly extract a numerical version of the duration (second + fraction) out of the Duration struct.

Comment (joshallen64) [reply PrimeRust/solution_4/PrimeRust.rs]: I corrected the output to remove the "s".

**Comment by joshallen64** [comment]
@rbergen I had to fix the docker file for the correct rust 1.51.  It broke when I updated it for the correct output format.  The method to get a float format for seconds in the duration class didn’t exist in rust 1.31.

**Review by rbergen — APPROVED** [review]

**Pull request merged**